### PR TITLE
refactor rename

### PR DESCRIPTION
**Pull request by Kishan-Dhakan**
### Changes
- Modify consensus logic similar to delete operation

### Tests
Tasks to complete before merging PR:
- [ ]  Ensure system tests are passing. If not [Run them manually](https://github.com/0chain/gosdk/actions/workflows/system_tests.yml) to check for any regressions :clipboard:
- [ ]  Do any new system tests need added to test this change? do any existing system tests need updated? If so create a PR at [0chain/system_test](https://github.com/0chain/system_test)
- [ ]  Merge your system tests PR to master AFTER merging this PR

### Associated PRs (Link as appropriate):
- blobber:
- 0chain:
- system_test:
- zboxcli:
- zwalletcli:
- Other: ...

### CODE REVIEW

File: zboxcore/sdk/renameworker.go [hunk 1]
```diff
@@ -121,7 +135,7 @@ func (req *RenameRequest) ProcessRename() error {
 		return fmt.Errorf("Rename failed: %s", err.Error())
 	}
 
-	req.consensus = 0
+	req.consensus.consensus = float32(removedNum)
```
Comment (lpoli) [inline zboxcore/sdk/renameworker.go]: req.consensus should be 0.
Previous was consensus on operation. Now it needs to be calculated for commit consensus.

And what is the logic behind using `removedNum` ?

Comment (Kishan-Dhakan) [reply zboxcore/sdk/renameworker.go]: Resolved

File: zboxcore/sdk/renameworker.go [hunk 2]
```diff
@@ -98,30 +98,33 @@ func (req *RenameRequest) ProcessRename() error {
...
 	}
 	err = writeMarkerMutex.Lock(context.TODO(), req.connectionID)
 	defer writeMarkerMutex.Unlock(context.TODO(), req.connectionID) //nolint: errcheck
 	if err != nil {
-		return fmt.Errorf("Rename failed: %s", err.Error())
+		return fmt.Errorf("rename failed: %s", err.Error())
 	}
 
-	req.consensus = 0
+	req.consensus.consensus = float32(0)
```
Comment (lpoli) [inline zboxcore/sdk/renameworker.go]: No need to type convert constants

**Review by lpoli — APPROVED** [review]

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | renameError  |
| gosdk  | renameRefactor  |
| zboxcli  | staging  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [success] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | renameError  |
| gosdk  | renameRefactor  |
| zboxcli  | staging  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Pull request merged**